### PR TITLE
enhancement(pulsar sink): fix missing drop count

### PR DESCRIPTION
**Pull request by prognant**
part of #14096
part of #13995

### CODE REVIEW

File: src/internal_events/pulsar.rs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+impl InternalEvent for PulsarSendingError {
+    fn emit(self) {
+        error!(
+            message = "Events dropped.",
+            reason = "A Pulsar sink generated an error.",
+            error = %self.error,
+            error_type = error_type::REQUEST_FAILED,
+            intentional = "false",
```
Comment (spencergilbert) [inline src/internal_events/pulsar.rs:19]: I thought we had changed these to booleans? Though I'm not sure it makes total sense, as we still need to convert them to strings for metrics.

File: src/internal_events/pulsar.rs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            stage = error_stage::SENDING,
+            count = %self.count,
+        );
+        counter!(
+            "component_errors_total", 1,
+            "error_type" => error_type::REQUEST_FAILED,
+            "stage" => error_stage::SENDING,
+        );
+
+        if self.count > 0 {
```
Comment (spencergilbert) [inline src/internal_events/pulsar.rs:29]: Is there a case where we'd error sending zero events?

Comment (prognant) [reply src/internal_events/pulsar.rs:29]: probably not, as calling `counter` with 0 is safe I will remove the condition.

**Review by spencergilbert — APPROVED** [review]

File: src/sinks/pulsar.rs [hunk 3]
```diff
@@ -356,9 +357,13 @@ impl Sink<Event> for PulsarSink {
                         protocol: "tcp",
                     });
                 }
-                Some((Err(error), _, finalizers)) => {
+                Some((Err(error), metadata, finalizers)) => {
                     finalizers.update_status(EventStatus::Errored);
                     error!(message = "Pulsar sink generated an error.", %error);
```
Comment (jszwedko) [inline src/sinks/pulsar.rs:362]: It seems like we should drop this `error!` since the event is also emitting a log.

Comment (spencergilbert) [reply src/sinks/pulsar.rs:362]: Good catch!

**Review by jszwedko — APPROVED** [review]

**Pull request merged**